pkg/catalog/routes_test.go

```diff
@@ -15,6 +16,56 @@ import (
 	"github.com/openservicemesh/osm/pkg/utils"
 )
 
+func TestRoutesFromRules(t *testing.T) {
+	assert := assert.New(t)
+	mc := MeshCatalog{meshSpec: smi.NewFakeMeshSpecClient()}
+
+	testCases := []struct {
+		name           string
+		rules          []target.TrafficTargetRule
+		namespace      string
+		expectedRoutes []trafficpolicy.HTTPRoute
+	}{
+		{
+			rules: []target.TrafficTargetRule{
+				{
+					Kind:    "HTTPRouteGroup",
+					Name:    tests.RouteGroupName,
+					Matches: []string{tests.BuyBooksMatchName},
+				},
+			},
+			namespace: tests.Namespace,
+			expectedRoutes: []trafficpolicy.HTTPRoute{
+				{
+					PathRegex: tests.BookstoreBuyPath,
+					Methods:   []string{"GET"},
+					Headers: map[string]string{
+						"user-agent": tests.HTTPUserAgent,
+					},
+				},
+			},
+		},
+		{
+			rules: []target.TrafficTargetRule{
+				{
+					Kind:    "HTTPRouteGroup",
+					Name:    "DoesNotExist",
+					Matches: []string{"hello"},
+				},
+			},
+			namespace:      tests.Namespace,
+			expectedRoutes: []trafficpolicy.HTTPRoute{},
+		},
+	}
+
+	for _, tc := range testCases {
+		routes, err := mc.routesFromRules(tc.rules, tc.namespace)
```

Comment:
It might be worth running each test case separately using `t.Run()` as done here https://github.com/openservicemesh/osm/blob/main/cmd/cli/metrics_test.go#L261

Reply:
Thanks!